cl/search/management/commands/sweep_indexer.py

```diff
@@ -0,0 +1,392 @@
+                        .order_by("pk")
+                    )
+                    q = [item.pk for item in queryset if item.is_judge]
+                    count = len(q)
+                    task_to_use = "index_parent_and_child_docs"
+                case "search.Opinion":
+                    queryset = (
+                        Opinion.objects.filter(pk__gte=last_document_id)
+                        .order_by("pk")
+                        .values_list("pk", "cluster_id")
+                    )
+                    count = queryset.count()
+                    q = queryset.iterator()
+                case "search.RECAPDocument":
+                    queryset = (
+                        RECAPDocument.objects.filter(pk__gte=last_document_id)
+                        .order_by("pk")
+                        .values_list("pk", "docket_entry__docket_id")
+                    )
+                    count = queryset.count()
+                    q = queryset.iterator()
+                case "audio.Audio":
+                    queryset = (
+                        Audio.objects.filter(
+                            pk__gte=last_document_id, processing_complete=True
+                        )
+                        .order_by("pk")
+                        .values_list("pk", flat=True)
+                    )
+                    count = queryset.count()
+                    q = queryset.iterator()
+                case _:
+                    # Base case for non parent-child documents type.
+                    model = apps.get_model(app_label)
+                    queryset = (
+                        model.objects.filter(pk__gte=last_document_id)
+                        .order_by("pk")
+                        .values_list("pk", flat=True)
+                    )
+                    count = queryset.count()
+                    q = queryset.iterator()
+
+            self.process_queryset(q, count, app_label, task_to_use)
+            #  After finishing each model, restart last_document_id to start
+            # from the ID 0 in the next model.
+            last_document_id = 0
+
+    def process_queryset(
+        self,
+        items: Iterable,
+        count: int,
+        app_label: str,
+        task_to_use: str,
+    ) -> None:
+        """Process a queryset and execute tasks based on the specified indexing
+        task_to_use.
+
+        :param items: Iterable of items to process. Items can be a simple
+        iterable of IDs or a tuple of (ID, changed_fields) for cases requiring
+        field changes.
+        :param count: Total number of items expected to process.
+        :param app_label: The app label and model that belongs to the queryset
+        being indexed.
+        :param task_to_use: The name of the celery task to execute.
+        :return: None
+        """
+        testing_mode = self.options.get("testing_mode", False)
+        model_options = {
```

Comment:
Could you combine this with the match statement above, so you just have one match statement or dictionary with all the data needed for each object?

Reply:
Yeah, I've moved the params within the match case statement in the previous method